azure-quantum/azure/quantum/target/microsoft/elements/dft/job.py

```diff
@@ -0,0 +1,34 @@
+from typing import Any, Dict
+from azure.quantum.job import JobFailedWithResultsError
+from azure.quantum.job.job import Job, DEFAULT_TIMEOUT
+from azure.quantum._client.models import JobDetails
+
+class MicrosoftElementsDftJob(Job):
+    """
+    A dedicated job class for jobs from the microsoft.dft target.
+    """
+
+    def __init__(self, workspace, job_details: JobDetails, **kwargs):
+        super().__init__(workspace, job_details, **kwargs)
+
+
+    def get_results(self, timeout_secs: float = DEFAULT_TIMEOUT) -> Dict[str, Any]:
+        try:
+            job_results = super().get_results(timeout_secs)
+            return job_results["results"]
+        except JobFailedWithResultsError as e:
+                failure_results = e.get_failure_results()
+                if "results" in failure_results \
```

Comment:
Does this line 21 cause any issues if failure_results is a string instead of a dict?

Reply:
Not this particular line since Python treats 
```python
 str1 in str2
```
as 
```python
str2.contains(str1)
```
but it could've very well failed in the next condition
```python
failure_results["results"]
```
if `failure_results` is a `str`. Added extensive validation to prevent error. Thanks!